libs/parallelism/algorithms/include/hpx/parallel/container_algorithms/rotate.hpp

```diff
@@ -122,26 +365,284 @@ namespace hpx { namespace parallel { inline namespace v1 {
+                hpx::is_execution_policy<ExPolicy>::value &&
+                hpx::traits::is_sentinel_for<Sent, FwdIter1>::value &&
+                hpx::traits::is_iterator<FwdIter2>::value
+            )>
+        // clang-format on
+        friend typename parallel::util::detail::algorithm_result<ExPolicy,
+            rotate_copy_result<FwdIter1, FwdIter2>>::type
+        tag_fallback_dispatch(hpx::ranges::rotate_copy_t, ExPolicy&& policy,
+            FwdIter1 first, FwdIter1 middle, Sent last, FwdIter2 dest_first)
+        {
+            static_assert((hpx::traits::is_forward_iterator<FwdIter1>::value),
+                "Requires at least forward iterator.");
+            static_assert((hpx::traits::is_forward_iterator<FwdIter2>::value),
+                "Requires at least forward iterator.");
+
+            typedef std::integral_constant<bool,
+                hpx::is_sequenced_execution_policy<ExPolicy>::value ||
+                    !hpx::traits::is_forward_iterator<FwdIter1>::value>
+                is_seq;
+
+            return hpx::parallel::v1::detail::rotate_copy<
+                rotate_copy_result<FwdIter1, FwdIter2>>()
+                .call2(std::forward<ExPolicy>(policy), is_seq(), first, middle,
+                    last, dest_first);
+        }
+
+        // clang-format off
+        template <typename Rng, typename OutIter,
+            HPX_CONCEPT_REQUIRES_(
+                hpx::traits::is_range<Rng>::value &&
+                hpx::traits::is_iterator<OutIter>::value
+            )>
+        // clang-format on
+        friend rotate_copy_result<
+            typename hpx::traits::range_iterator<Rng>::type, OutIter>
+        tag_fallback_dispatch(hpx::ranges::rotate_copy_t, Rng&& rng,
+            typename hpx::traits::range_iterator<Rng>::type middle,
+            OutIter dest_first)
+        {
+            return hpx::parallel::v1::detail::rotate_copy<rotate_copy_result<
+                typename hpx::traits::range_iterator<Rng>::type, OutIter>>()
+                .call(hpx::execution::seq, hpx::util::begin(rng), middle,
+                    hpx::util::end(rng), dest_first);
+        }
+
+        // clang-format off
+        template <typename ExPolicy, typename Rng, typename OutIter,
+            HPX_CONCEPT_REQUIRES_(
+            hpx::is_execution_policy<ExPolicy>::value&& hpx::traits::is_range<
+                Rng>::value&& hpx::traits::is_iterator<OutIter>::value)>
+        // clang-format on
+        friend typename parallel::util::detail::algorithm_result<ExPolicy,
+            rotate_copy_result<typename hpx::traits::range_iterator<Rng>::type,
+                OutIter>>::type
+        tag_fallback_dispatch(hpx::ranges::rotate_copy_t, ExPolicy&& policy,
+            Rng&& rng, typename hpx::traits::range_iterator<Rng>::type middle,
+            OutIter dest_first)
+        {
+            return hpx::parallel::v1::detail::rotate_copy<rotate_copy_result<
```

Comment:
Add typedef std::integral_constant<bool,
                hpx::is_sequenced_execution_policy<ExPolicy>::value ||
                    !hpx::traits::is_forward_iterator<typename hpx::traits::range_iterator<Rng>::type>::value>
                is_seq;

Reply:
So you mean the following:

`typedef std::integral_constant<bool,
                hpx::is_sequenced_execution_policy<ExPolicy>::value ||
                    !hpx::traits::is_bidirectional_iterator<
                        typename hpx::traits::range_iterator<Rng>::type>::value>
                is_seq;`
the plain text always miss sth.

Reply:
Yep, sorry should've used the code block while pasting.